### PR TITLE
xssi-prefix support to amp-form

### PR DESCRIPTION
**Pull request by samouri**
Addresses `b/143896641`

cc @choumx / @jridgewell

### CODE REVIEW

**Comment by samouri** [comment]
@choumx  / @jridgewell: could you give this a review when you have spare cycles? All the tests are passing according to travis (but not according to the X marks in the GH display).

File: extensions/amp-form/0.1/amp-form.js [hunk 1]
```diff
@@ -1000,7 +1000,9 @@ export class AmpForm {
     let promise;
     if (e && e.response) {
       const error = /** @type {!Error} */ (e);
-      promise = error.response.json().catch(() => null);
+      promise = this.xhr_
+        .xssiJson(error.response, this.form_.getAttribute('xssi-prefix'))
```
Comment (choumx) [inline extensions/amp-form/0.1/amp-form.js]: Nit: Dedupe the attribute name.

File: extensions/amp-form/amp-form.md [hunk 2]
```diff
@@ -129,6 +129,11 @@ The value for `action-xhr` can be the same or a different endpoint than `action`
 
 To learn about redirecting the user after successfully submitting the form, see the [Redirecting after a submission](#redirecting-after-a-submission) section below.
 
+##### xssi-prefix (optional)
+
+Specifies that a prefix should be stripped prior to parsing the fetched json from the `action-xhr` endpoint. If the prefix is not present in the response, then this attribute will have no effect.
+This can be useful for APIs that include [security prefixes](http://patorjk.com/blog/2013/02/05/crafty-tricks-for-avoiding-xssi/) like `)]}` to help prevent cross site scripting attacks.
+
 ##### data-initialize-from-url
```
Comment (choumx) [inline extensions/amp-form/amp-form.md]: Minor: I think this should be "(optional)" too. Then I'd move `xssi-prefix` below it.

File: extensions/amp-form/0.1/amp-form.js (hunk 1 above)
Comment (samouri) [reply extensions/amp-form/0.1/amp-form.js]: Do you mean extract `'xssi-prefix'` into a class-level variable?

File: extensions/amp-form/amp-form.md (hunk 2 above)
Comment (samouri) [reply extensions/amp-form/amp-form.md]: Done

**Pull request merged**